crates/tests/implement_core/src/static_com_object.rs

```diff
@@ -0,0 +1,77 @@
+//! Unit tests for `windows_core::StaticComObject`
+
+use std::sync::atomic::{AtomicU32, Ordering::SeqCst};
+use windows_core::{
+    implement, interface, ComObject, IUnknown, IUnknownImpl, IUnknown_Vtbl, InterfaceRef,
+    StaticComObject,
+};
+
+#[interface("818f2fd1-d479-4398-b286-a93c4c7904d1")]
+unsafe trait INumberFactory: IUnknown {
+    fn next(&self) -> u32;
+
+    fn add(&self, x: u32, y: u32) -> u32;
+}
+
+#[implement(INumberFactory)]
+struct MyFactory {
+    x: AtomicU32,
+}
+
+impl INumberFactory_Impl for MyFactory_Impl {
+    unsafe fn next(&self) -> u32 {
+        self.x.fetch_add(1, SeqCst)
+    }
+
+    unsafe fn add(&self, x: u32, y: u32) -> u32 {
+        x + y
+    }
+}
+
+static NUMBER_FACTORY_INSTANCE: StaticComObject<MyFactory> = MyFactory {
+    x: AtomicU32::new(100),
+}
+.into_static();
+
```

Comment:
Ideally there would be consistency here with `ComObject` where you'd use `new`. Just makes it a little harder to learn I have to use `new` for ComObject but `into_static` for `StaticComObject`.

Reply:
Something we can hopefully improve in future.